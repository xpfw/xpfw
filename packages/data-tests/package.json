{
  "name": "@xpfw/data-tests",
  "version": "1.0.0-beta.16",
  "main": "dist",
  "typings": "dist",
  "license": "MIT",
  "peerDependencies": {
    "@xpfw/form": "1.x.x",
    "@xpfw/data": "1.x.x",
    "@xpfw/data-feathers": "1.x.x",
    "ajv": "^6.x.x",
    "lodash-es": "^4.x.x",
    "mobx": "^5.x.x",
    "react": "^16.x.x"
  },
  "scripts": {
    "build": "./node_modules/.bin/rollup -c",
    "dev": "./node_modules/.bin/rollup -c -w",
    "test": "./node_modules/.bin/jest --ci",
    "update": "ncu -a",
    "prepare": "npm run build"
  },
  "devDependencies": {
    "lodash": "^4.17.11",
    "lodash-es": "^4.17.11",
    "mobx": "^5.9.0",
    "ajv": "^6.10.0",
    "moment": "^2.24.0",
    "@types/jest": "^24.0.11",
    "@types/lodash": "^4.14.123",
    "@types/lodash-es": "^4.17.3",
    "@types/mockdate": "^2.0.0",
    "@types/react": "^16.8.8",
    "@types/socket.io-client": "^1.4.32",
    "@types/react-test-renderer": "^16.8.1",
    "@xpfw/test-util": "1.0.0-beta.16",
    "@xpfw/validate": "^0.1.2",
    "@xpfw/form": "^1.0.0-beta.16",
    "@xpfw/form-tests": "^1.0.0-beta.16",
    "@xpfw/form-shared": "^0.1.5",
    "@xpfw/data": "^1.0.0-beta.16",
    "@xpfw/data-feathers": "^1.0.0-beta.16",
    "jest": "^24.5.0",
    "jest-cli": "^24.5.0",
    "mockdate": "^2.0.2",
    "react": "^16.8.5",
    "react-test-renderer": "^16.8.5",
    "rollup": "^1.7.3",
    "rollup-plugin-typescript2": "^0.20.1",
    "rollup-plugin-uglify": "^6.0.2",
    "ts-jest": "=24.0.0",
    "tslib": "^1.9.3",
    "typescript": "^3.3.4000",
    "uglify-es": "^3.3.10"
  },
  "jest": {
    "globals": {
      "ts-jest": {
        "diagnostics": false
      }
    },
    "transform": {
      ".(ts|tsx)": "ts-jest"
    },
    "testPathIgnorePatterns": [
      "\\.snap$",
      "<rootDir>/node_modules/",
      "<rootDir>/lib",
      "<rootDir>/dist"
    ],
    "moduleNameMapper": {
    },
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "testURL": "http://localhost/"
  }
}
