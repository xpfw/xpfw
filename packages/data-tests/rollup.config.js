// rollup.config.js
import typescript from "rollup-plugin-typescript2"
import uglify from "rollup-plugin-uglify"
import { minify } from "uglify-es"

export default {
  input: 'src/index.ts',
  output: {
    file: 'dist/index.js',
    format: 'cjs'
  },
  plugins: [
    typescript(),
    // uglify({}, minify)
  ],
  external: [
    "isomorphic-fetch",
    "ajv",
    "lodash",
    "lodash-es",
    "mobx",
    "react",
    "react-dom",
    "preact-render-to-json",
    "moment",
    "mockdate",
    "@xpfw/form",
    "@xpfw/form-tests",
    "@xpfw/form-shared",
    "@xpfw/ui-shared",
    "moment",
    "@feathersjs/feathers",
    "@feathersjs/authentication-client",
    "@feathersjs/socketio-client",
    "@feathersjs/rest-client",
    "resub",
    "resub-persist",
    "socket.io-client",
    "@xpfw/validate",
    "@xpfw/form-shared",
    "@xpfw/ui-tests",
    "@xpfw/ui-feathers",
    "@xpfw/data",
    "@xpfw/data-tests",
    "@xpfw/data-feathers",
    "@xpfw/test-util",
    "react",
    "react-dom",
    "react-test-renderer",
    "isomorphic-fetch"
  ]
};