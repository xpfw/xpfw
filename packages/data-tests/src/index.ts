import authTest from "./tests/auth"
import createTest from "./tests/create"
import editTest from "./tests/edit"
import listTest from "./tests/list"
import relationshipTest from "./tests/relationship"
import removeTest from "./tests/remove"
import showTest from "./tests/show"
import toJS from "./testUtil/toJS"

export {
  authTest, createTest, editTest, removeTest, showTest, toJS,
  listTest, relationshipTest
}
