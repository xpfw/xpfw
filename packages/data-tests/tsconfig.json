{

  "compilerOptions": {
    "outDir": "dist",
    "module": "es2015",
    "target": "es2015",
    "lib": ["es6", "dom"],
    "sourceMap": true,
    "allowJs": false,
    "jsx": "react",
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "declaration": true,
    "strict": true,
    "baseUrl": ".",
    "paths": {
        "*": ["*"]
    },
    "typeRoots": [
      "./@types",
      "./src/@types",
      "./node_modules/@types"
    ]
  },
  "exclude": [
    "node_modules",
    "dist",
    "scripts",
    "acceptance-tests",
    "webpack",
    "jest",
    "src/setupTests.ts",
    "./src/scripts/",
    "src/testUtil/**/*.ts",
    "src/testUtil/**/*.tsx",
    "**/*.test.ts",
    "**/*.test.tsx"
  ],
  "include": [
    "./src/"
  ],
  "types": [
    "typePatches"
  ]
}
