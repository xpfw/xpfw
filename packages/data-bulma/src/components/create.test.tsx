import { createTest } from "@xpfw/data-tests"
import "isomorphic-fetch"
import BulmaCreate from "./create"

createTest(BulmaCreate)
