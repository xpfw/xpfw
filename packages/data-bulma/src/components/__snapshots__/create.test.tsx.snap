// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DbStore Create Test: Before reset on mount 1`] = `
FormStoreClass {
  "errors": Object {
    "formModel": Object {
      "errors": Array [
        "a",
        "b",
      ],
    },
  },
  "formData": Object {
    "authForm": undefined,
    "formModel": Object {
      "guestName": "myText",
      "myVal": 420,
    },
  },
  "loading": Object {
    "formModel": false,
  },
}
`;

exports[`DbStore Create Test: Before reset on mount 2`] = `
FormStoreClass {
  "errors": Object {
    "createpref": Object {
      "formModel": Object {
        "errors": Array [
          "a",
          "b",
        ],
      },
    },
    "formModel": Object {
      "errors": Array [
        "a",
        "b",
      ],
    },
  },
  "formData": Object {
    "authForm": undefined,
    "createpref": Object {
      "formModel": Object {
        "guestName": "SECOND OBJECT",
        "myVal": 421,
      },
    },
    "formModel": Object {
      "guestName": "myText",
      "myVal": 420,
    },
  },
  "loading": Object {
    "createpref": false,
    "formModel": false,
  },
}
`;

exports[`DbStore Create Test: after create 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="myText"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={420}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-success"
  >
    Successfully created 
    0
  </div>
</div>
`;

exports[`DbStore Create Test: after login 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
</div>
`;

exports[`DbStore Create Test: after purposely set invalid err 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="myText"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={420}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: after reset on mount 1`] = `
FormStoreClass {
  "errors": Object {
    "formModel": Object {
      "errors": Array [
        "a",
        "b",
      ],
    },
  },
  "formData": Object {
    "authForm": undefined,
    "formModel": Object {
      "guestName": "myText",
      "myVal": 420,
    },
  },
  "loading": Object {
    "formModel": false,
  },
}
`;

exports[`DbStore Create Test: after reset on mount 2`] = `
FormStoreClass {
  "errors": Object {
    "createpref": Object {
      "formModel": Object {
        "errors": Array [
          "a",
          "b",
        ],
      },
    },
    "formModel": Object {
      "errors": Array [
        "a",
        "b",
      ],
    },
  },
  "formData": Object {
    "authForm": undefined,
    "createpref": Object {
      "formModel": Object {
        "guestName": "SECOND OBJECT",
        "myVal": 421,
      },
    },
    "formModel": Object {
      "guestName": "myText",
      "myVal": 420,
    },
  },
  "loading": Object {
    "createpref": false,
    "formModel": false,
  },
}
`;

exports[`DbStore Create Test: before create 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="myText"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={420}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
</div>
`;

exports[`DbStore Create Test: before login 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
</div>
`;

exports[`DbStore Create Test: hidden in create 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="myText"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={420}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: hidden text 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="myText"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={420}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: not hidden in create 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="myText"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={420}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: prefixed after create 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="SECOND OBJECT"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={421}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
</div>
`;

exports[`DbStore Create Test: prefixed after purposely set invalid err 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="SECOND OBJECT"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={421}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: prefixed before create 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
</div>
`;

exports[`DbStore Create Test: prefixed hidden in create 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="SECOND OBJECT"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={421}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: prefixed hidden text 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="SECOND OBJECT"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={421}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: prefixed not hidden in create 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="SECOND OBJECT"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={421}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: prefixed while loading 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="SECOND OBJECT"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={421}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
</div>
`;

exports[`DbStore Create Test: resetOnMount = true 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="myText"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={420}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: resetOnMount = true 2`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="myText"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={420}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-danger"
  >
    Error please recheck your inputs or connection 
    {"errors":["a","b"]}
  </div>
</div>
`;

exports[`DbStore Create Test: while loading 1`] = `
<div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      guestName
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="text"
      value="myText"
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      myVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="number"
      value={420}
    />
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      dateVal
    </label>
    <input
      className="input"
      onChange={[Function]}
      type="date"
      value="2011-05-05"
    />
  </div>
  <a
    className="button is-primary"
    onClick={[Function]}
  >
    Create
  </a>
  <div
    className="notification is-success"
  >
    Successfully created 
    0
  </div>
</div>
`;
